core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -113,12 +129,12 @@ private[history] class FsHistoryProvider(conf: SparkConf) extends ApplicationHis
         "Logging directory specified is not a directory: %s".format(logDir))
     }
 
-    checkForLogs()
+    // A task that periodically checks for event log updates on disk.
+    pool.scheduleAtFixedRate(getRunner(checkForLogs), 0, UPDATE_INTERVAL_MS, TimeUnit.MILLISECONDS)
 
-    // Disable the background thread during tests.
-    if (!conf.contains("spark.testing")) {
```

Comment:
I don't know why this was turned off for testing, but it doesn't seem like you want to change that behavior, do you?